app/src/main/kotlin/com/softteco/template/ui/feature/resetPassword/ResetPasswordScreen.kt

```diff
@@ -0,0 +1,209 @@
+package com.softteco.template.ui.feature.resetPassword
+
+import androidx.compose.animation.animateColorAsState
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Create
+import androidx.compose.material.icons.filled.Done
+import androidx.compose.material.icons.rounded.Check
+import androidx.compose.material.icons.rounded.Close
+import androidx.compose.material3.Button
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.softteco.template.R
+import com.softteco.template.ui.components.TextSnackbarContainer
+import com.softteco.template.ui.feature.PasswordFieldState
+import com.softteco.template.ui.theme.AppTheme
+import com.softteco.template.ui.theme.Dimens
+
+@Composable
+fun ResetPasswordScreen(
+    token: String?,
+    modifier: Modifier = Modifier,
+    gotToLoginScreen: () -> Unit = {},
+    viewModel: ResetPasswordViewModel = hiltViewModel()
+) {
+    val state by viewModel.state.collectAsState()
+    ScreenContent(
+        token = token,
+        state = state,
+        modifier = modifier,
+        gotToLoginScreen = gotToLoginScreen,
+    )
+}
+
+@Composable
+private fun ScreenContent(
+    token: String?,
+    state: ResetPasswordViewModel.State,
+    modifier: Modifier = Modifier,
+    gotToLoginScreen: () -> Unit = {}
```

Comment:
Looks like there's a typo here)

Reply:
No) it is not a typo. If request for restoring password is seccessed - we go to Login screen

Reply:
You meant extra letter)